tests/unit/test_parts.py

```diff
@@ -407,17 +407,27 @@ def test_part_unmarshal_type_error(self, partitions):
         with pytest.raises(errors.PartSpecificationError) as raised:
             Part("foo", {"plugin": []}, partitions=partitions)
         assert raised.value.part_name == "foo"
-        assert raised.value.message == "- str type expected in field 'plugin'"
+        assert (
+            raised.value.message == "- Input should be a valid string in field 'plugin'"
+        )
+
+    @pytest.mark.parametrize("fileset", ["stage", "prime"])
+    def test_relative_path_validation_slash(self, partitions, fileset):
+        with pytest.raises(errors.PartSpecificationError) as raised:
+            Part("foo", {fileset: ["bar", "/baz", "x"]}, partitions=partitions)
+        assert raised.value.part_name == "foo"
+        assert raised.value.message == (
+            "- Value error, '/baz' must be a relative path (cannot start with "
+            f"'/') in field '{fileset}'"
+        )
 
     @pytest.mark.parametrize("fileset", ["stage", "prime"])
-    def test_relative_path_validation(self, partitions, fileset):
+    def test_relative_path_validation_empty(self, partitions, fileset):
```

Comment:
how come this was split into two tests?

Reply:
because `field_validator` decorator, the replacement for the deprecated `validator`, doesn't support `each_item` property. So instead of checking each item in the list with `each_item` automatically, my alternative solution was to check them manually with for loop, which leads to only one malformed field raising an error even in the case of multiple malformed fields. For that purpose, I had to make a separate test for each of the two malformations: one for the empty value and another for the slash-starting value.

I think there should be a way to replicate the old behavior of `each_item` with using Annotated, but not sure if it will need to a different outcome. You can check the current "hacky" implementation here:
https://github.com/linostar/craft-parts/blob/6369ff3d193c35cf21c33035c18b378c6a6a396f/craft_parts/parts.py#L97

Reply:
ah I see. That's a minor-ish regression, because it means that we won't
report all validation errors in one go. We should preserve the original
behavior if possible